src/bluesky/run_engine.py

```diff
@@ -269,7 +270,7 @@ class RunEngine:
         completely up to the user.
         Expected return: normalized metadata
 
-    scan_id_source : callable, optional
```

Comment:
Please leave as optional and update to note that the function can be sync or async.